var/spack/repos/builtin/packages/tasmanian/package.py

```diff
@@ -30,7 +30,94 @@ class Tasmanian(CMakePackage):
+    # tested with OpenMP 3.1 (clang4) through 4.0-4.5 (gcc 5 - 8)
+
+    variant('blas', default=False,
+            description='add BLAS support to Tasmanian')
+
+    variant('mpi', default=False,
+            description='add MPI support to Tasmanian')
+
+    variant('cuda', default=False,
+            description='add CUDA support to Tasmanian')
+
+    variant('magma', default=False,
+            description='add UTK MAGMA support to Tasmanian')
+
+    variant('python', default=False,
+            description='add Python binding for Tasmanian')
+
+    depends_on('cmake@3.5.0:', type='build')
+
+    depends_on('python@2.7:', when='+python', type=('build', 'run'))
+    depends_on('py-numpy', when='+python', type=('build', 'run'))
+
+    extends('python', when='+python')
+
+    depends_on('mpi', when="+mpi")  # openmpi 2 and 3 tested
+
+    depends_on('blas', when="+blas")  # openblas 0.2.18 or newer
+
+    depends_on('cuda@9.1', when='+cuda', type=('build', 'run'))
+    depends_on('cuda@9.1', when='+magma', type=('build', 'run'))
+
+    depends_on('magma@2.3.0', when='+magma', type=('build', 'run'))
+
+    def cmake_args(self):
+        spec = self.spec
+
+        if '+xsdkflags' in spec:
+            args = [
+                '-DUSE_XSDK_DEFAULTS:BOOL=ON',
+                '-DXSDK_ENABLE_PYTHON:BOOL={0}'.format(
+                    'ON' if '+python' in spec else 'OFF'),
+                '-DTasmanian_ENABLE_MPI:BOOL={0}'.format(
+                    'ON' if '+mpi' in spec else 'OFF'),
+                '-DXSDK_ENABLE_OPENMP:BOOL={0}'.format(
+                    'ON' if '+openmp' in spec else 'OFF'),
+                '-DTPL_ENABLE_BLAS:BOOL={0}'.format(
+                    'ON' if '+blas' in spec else 'OFF'),
+                '-DXSDK_ENABLE_CUDA:BOOL={0}'.format(
+                    'ON' if '+cuda' in spec else 'OFF'),
+                '-DXSDK_ENABLE_CUDA:BOOL={0}'.format(
+                    'ON' if '+magma' in spec else 'OFF'),
+                '-DTPL_ENABLE_MAGMA:BOOL={0}'.format(
+                    'ON' if '+magma' in spec else 'OFF'), ]
+        else:
+            args = [
+                '-DTasmanian_ENABLE_OPENMP:BOOL={0}'.format(
+                    'ON' if '+openmp' in spec else 'OFF'),
+                '-DTasmanian_ENABLE_BLAS:BOOL={0}'.format(
+                    'ON' if '+blas' in spec else 'OFF'),
+                '-DTasmanian_ENABLE_PYTHON:BOOL={0}'.format(
+                    'ON' if '+python' in spec else 'OFF'),
+                '-DTasmanian_ENABLE_MPI:BOOL={0}'.format(
+                    'ON' if '+mpi' in spec else 'OFF'),
+                '-DTasmanian_ENABLE_CUDA:BOOL={0}'.format(
+                    'ON' if '+cuda' in spec else 'OFF'),
+                '-DTasmanian_ENABLE_CUDA:BOOL={0}'.format(
+                    'ON' if '+magma' in spec else 'OFF'),
+                '-DTasmanian_ENABLE_MAGMA:BOOL={0}'.format(
+                    'ON' if '+magma' in spec else 'OFF'), ]
```

Comment:
I think most of these two sections are duplicates. You should be able to refractor all of this logic now.

Reply:
To be a member of xSDK, I have to support two `cmake` modes with different naming conventions:
* if `USE_XSDK_DEFAULTS=ON` then use `XSDK_ENABLE_PYTHON=ON`
* if `USE_XSDK_DEFAULTS=OFF` then use `Tasmanian_ENABLE_PYTHON=ON`
* to avoid conflicts, Tasmanian accepts only one type of options at a time and does internal remapping

I am not sure if @balay or someone else from xSDK is following this dicussion, but I cannot change the spack without also changing the cmake build system and violating a good number of policies below:

https://figshare.com/articles/xSDK_Community_Installation_Policies_GNU_Autoconf_and_CMake_Options/4495133

Reply:
Ah, sorry, I didn't notice the `XSDK` vs. `Tasmanian` distinction.